### DISCUSSION TITLE
Demos Commander (file manager)

### DISCUSSION THREAD

**Discussion by ATroubleshooter** [jbruchon/elks#1754] [Ideas]
Hello everyone!

There's such a thing as Demos Commander https://github.com/sergev/deco
It's development was dropped in late '90s.

![image](https://github.com/ghaerr/elks/assets/30573655/50976069-44ae-4106-a90c-c0821d72a823)

I wonder if it can be easily ported to ELKS?
Thank you

**Comment by ghaerr** [comment]
A quick look shows `deco` looks probably fairly straightforward to port, termcap based. The UI is nice. I can't tell how big it is, or whether the data segment might be large.

I'm not sure its worth the effort - I wrote the `fm` file manager a year ago, and I think very few use it. It does mostly the same thing except not as nice looking:
<img width="832" alt="fm" src="https://github.com/ghaerr/elks/assets/11985637/d0e85b98-a7b3-4dd1-9bba-a11f289496a5">

If someone wants to take a pass at porting `deco`, I will help with questions. What do you think - have you ever tried `fm`, and just want something fancier?

**Reply by ATroubleshooter** [reply]
I tried fm, and I'm using it from time to time.

Thank you, @ghaerr

**Comment by mezantrop** [comment]
I tried to revive `deco` a few years ago https://gitlab.com/mezantrop/deco, perhaps, you may find something useful there. And here is the repo from the developer https://github.com/sergev/deco, he seem to start working on it again after all those years :)

**Comment by floriangit** [comment]
`walk` which I also use often on Linux is similar to `fm`, but even more UNIX-like with cool features.
I looked quickly but discovered it's written in GOlang. Then I looked (again briefly) at ghaerr's `fm.c` and - for now - gave up. To marry both sources might be quite some work?

https://github.com/antonmedv/walk

**Comment by ghaerr** [comment]
I would say trying to merge `fm` and `walk` would be 1) lots of work, because `walk` is written in go, and 2) not really worth it, since `fm` is very similar; that is, displays filenames with a minimum of extra decorations. `fm` could more easily be extended to display non-text filename contents, should anyone really ever use it.

On the other hand, `deco` seems much more user-friendly, and matches some far less portable programs (for ELKS) like Midnight Commander. I took a brief walkthrough of `deco` source code and think it could be somewhat easily ported to ELKS. 

The bigger question is - does/would anyone ever really use such programs on ELKS?

**Comment by floriangit** [comment]
Thanks for the feedback.

> The bigger question is - does/would anyone ever really use such programs on ELKS?

An even bigger question: who is using ELKS - aside from hobbyists - for productive usages? Do we know?
I'm just a hobbyist ;-)

**Reply by toncho11** [reply]
ELKS is a fun project! I think this answers everything.
